server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -937,9 +937,15 @@ public boolean isAggregatable() {
             return fielddata;
         }
 
+        public boolean canUseSyntheticSourceDelegate() {
+            return syntheticSourceDelegate != null
+                && syntheticSourceDelegate.ignoreAbove() == Integer.MAX_VALUE
+                && syntheticSourceDelegate.hasNormalizer() == false;
+        }
```

Comment:
A synthetic _source delegate will never have a normalizer. It might have `ignore_above` though. If it does I think it's still safe to use it for fetching but it won't help at the moment.

Reply:
This method should probably have a name like `canUseSyntheticSourceDelegateForQuerying`.

Reply:
Oh! There's no use in using the delegate for querying if the field isn't indexed. maybe check that?

And! I'm not 100% sure block loading works for synthetic source keyword fields with doc values disabled. We *should* fall to the `originalName()` stored field. If we don't that's a bug.

Reply:
> And! I'm not 100% sure block loading works for synthetic source keyword fields with doc values disabled. We should fall to the originalName() stored field. If we don't that's a bug.

I'm not sure I understand the exact use case here. I tried a few cases with KEYWORD fields alone and with TEXT fields with KEYWORD subfields, but every time I try to disable doc_values, I get an error like `field .. doesn't support synthetic source unless it is stored or has a sub-field of type [keyword] with doc values or stored and without a normalizer`.

Do you have an example?

Reply:
Sorry -it looks like you need both `doc_values: false, stored: true, ignore_above: 12` or something. I think it's probably just an issue to file and run down later.